### PR TITLE
fix out-of-place version of FineRKN5

### PR DESCRIPTION
**Pull request by ranocha**
I found some typos in out-of-place version of `FineRKN5()` and fixed them. I also added some tests checking that the in-place and out-of-place versions of some Runge-Kutta-Nyström methods. Some of these are currently broken, in particular with adaptive time stepping. I propose to open an issue and check them later.
I also fixed the reported number of function evaluations of some methods and added some checks for them.

CC @HenryLangner

### CODE REVIEW

File: src/perform_step/rkn_perform_step.jl
Comment (HenryLangner) [inline src/perform_step/rkn_perform_step.jl:147]: I missed that in the first implementation https://github.com/SciML/OrdinaryDiffEq.jl/pull/1948, thank you for fixing it!

**Review by HenryLangner — APPROVED** [review]
Something was off with the "out-of-place" version of ```FineRKN5```. I missed these mistakes, thank you for fixing them!

I will add convergence test for the ```damped_oscillator``` in another PR.

**Comment by ranocha** [comment]
I don't think the failing downstream tests are related to this PR - but they fail badly, e.g.,
https://github.com/SciML/OrdinaryDiffEq.jl/actions/runs/5309278846/jobs/9609731280?pr=1973#step:6:1413
```
signal (4): Illegal instruction
in expression starting at /home/runner/work/OrdinaryDiffEq.jl/OrdinaryDiffEq.jl/downstream/test/sde_neural.jl:13
_pullback at /home/runner/work/OrdinaryDiffEq.jl/OrdinaryDiffEq.jl/downstream/test/sde_neural.jl:104 [inlined]
_pullback at /home/runner/.julia/packages/Zygote/JeHtr/src/compiler/interface2.jl:0
_pullback at /home/runner/work/OrdinaryDiffEq.jl/OrdinaryDiffEq.jl/downstream/test/sde_neural.jl:133 [inlined]
_pullback at /home/runner/.julia/packages/Zygote/JeHtr/src/compiler/interface2.jl:0
```

**Comment by ChrisRackauckas** [comment]
Was there a necessary order here? I merged the one I saw that looked good to go, so I think this just needs a quick merge resolution to be merged next? https://github.com/SciML/OrdinaryDiffEq.jl/pull/1975 then has a few conflicts so it can be merged shortly after.

**Comment by HenryLangner** [comment]
> It was messed up a bit since @HenryLangner pulled the changes I made here into his PR. But it's fine, I fixed the merge conflicts here. I propose to merge this first and let @HenryLangner resolve the merge conflicts in #1975 afterwards, updating also the new tests added here with the new algorithm.

Sounds good. Once this PR is merged I will update my PR https://github.com/SciML/OrdinaryDiffEq.jl/pull/1975, resolve merge conflicts and update the new tests.

**Pull request merged**
